src/realm/query_engine.cpp

```diff
@@ -856,15 +876,41 @@ size_t LinksToNode<NotEqual>::find_first_local(size_t start, size_t end)
     REALM_ASSERT(m_target_keys.size() == 1);
     ObjKey key = m_target_keys[0];
 
-    if (m_column_type == col_type_LinkList || m_condition_column_key.is_set()) {
-        BPlusTree<ObjKey> links(m_table.unchecked_ptr()->get_alloc());
-        for (size_t i = start; i < end; i++) {
-            if (ref_type ref = get_ref(i)) {
-                links.init_from_ref(ref);
-                auto sz = links.size();
-                for (size_t j = 0; j < sz; j++) {
-                    if (links.get(j) != key) {
+    if (m_condition_column_key.is_collection()) {
```

Comment:
This code seems the same of the one that starts at line 822, maybe this can be encapsulated into some utility method.

Reply:
It probably could, but I am not sure the code would be much nicer to read.